### PR TITLE
Add serialization support for torch.jit.ScriptModule instances

### PR DESCRIPTION
**Pull request by midokura-silvia**
This pull request modifies the serde module to support serialization for torch.jit.ScriptModule instances. 
This covers both functions decorated with @torch.jit.script (torch.jit.ScriptModule) and models traced using torch.jit.trace() (torch.jit.TopLevelTracedModule).

### CODE REVIEW

**Comment by midokura-silvia** [comment]
Travis fails due to bug in torch version 1.1.0

**Comment by robert-wagner** [comment]
> Travis fails due to bug in torch version 1.1.0

What exactly is the bug?

**Comment by iamtrask** [comment]
> Travis fails due to bug in torch version 1.1.0

This is surprising since I thought Travis built using Torch version 1.0.1

**Comment by LaRiffle** [comment]
> > Travis fails due to bug in torch version 1.1.0
> 
> This is surprising since I thought Travis built using Torch version 1.0.1

My bad Andrew I told you smthg wrong, Travis will fetch the latest version (but with requirement >=1.0.1) so it will get 1.1

File: test/test_serde.py [hunk 1]
```diff
@@ -612,3 +599,66 @@ def test_serde_object_wrapper_int():
...
+    obj_wrapper = pointers.ObjectWrapper(obj, id=200)
+    msg = serde.serialize(obj_wrapper)
+
+    obj_wrapper_received = serde.deserialize(msg)
+
+    pred_before = obj(data)
+
+    pred_after = obj_wrapper_received.obj(data)
+
+    assert pred_before.sum() == pred_after.sum()
```
Comment (mari-linhares) [inline test/test_serde.py]: would be better to compare if all elements match

File: test/workers/test_virtual.py [hunk 2]
```diff
@@ -226,3 +228,18 @@ def test_spinup_time(hook):
     dummy = sy.VirtualWorker(hook, id="dummy", data=data)
     end_time = time()
     assert (end_time - start_time) < 0.05
+
+
+@pytest.mark.skip(reason="bug in pytorch version 1.1.0, jit.trace returns raw C function")
```
Comment (robert-wagner) [inline test/workers/test_virtual.py]: If you're going to skip this function, you also need to mark it with #pragma: no cover

**Pull request merged**